src/ml-search.service.js

```diff
@@ -1135,11 +1135,55 @@
       var self = this;
       
       // TODO: what other conditions
+      // [GJo] This looks a bit peculiar. If you have start and limit in params, but no options, then they get lost?
       if ( !options && params && params.options) {
         options = params;
         params = null;
       }
```

Comment:
Optional arguments (note the brackets around the argument names in the doc comments).